### PR TITLE
Added zpool compatibility

### PR DESCRIPTION
**Pull request by hacor**
All options work through:

var zfs-filesystem = require('zfs');

for zfs command:
zfs-filesystem.zfs.get ...

(The old commands are still available without the zfs middle command: zfs-filesystem.get)

for zpool command:
zfs-filesystem.zpool.get ...

Everything roughly tested, but everything seems to work.

Best regards!

Hans

### CODE REVIEW

**Comment by calmh** [comment]
Thanks! Some comments; the compact and findCmd functions might be better in a common until.js or so instead of being duplicated? Can we capitalize ZPOOL as ZPool?

**Comment by hacor** [comment]
All remarks resolved!
Greetings!

File: lib/zpool.js [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    obj.health = info[5];
+    if (info[6] !== '-') {
+        obj.altroot = info[6];
+    }
+
+    Object.freeze(obj);
+    return obj;
+}
+
+function analyzeVdevOpts (deviceList) {
```
Comment (calmh) [inline lib/zpool.js]: This should probably be renamed, if it deserves to exist on its own? It does not actually analyze any vdev opts. :)

**Comment by calmh** [comment]
Looks reasonable apart from the above nit and not being auto-mergeable; can you rebase onto current master?

**Comment by hacor** [comment]
Calmh,

The name indeed didn't made sense anymore. I created this function with a more advanced set of options in mind: The end user picks a poolType like raid, mirror, ...: and this function puts everything in the correct order for the zpool command. So the opts param would look like:

opts.devices = "['/dev/sda/', '/dev/sdb/']"
opts.poolType = "mirror"

Then the vdev analyze function would create this output: 
mirror /dev/sda mirror /dev/sdb

But anyway it's not in for the moment, so I renamed the function. When I've some more time in the future I'll finish it. ;-)

I'm looking into rebasing, but it can take some time before a create a new pull request. It's the first time I'm active on a github project, so I've a lot to learn :-). I'm looking for some good rebasing tutorials right now.

Anyway thanks for the advice and the opportunity to make contributions!

Best regards!

H

**Comment by hacor** [comment]
I think rebasing worked

**Comment by hacor** [comment]
sorry, one exports got lost during rebasing. Now the tests pass successfully.

Thanks for your patience!

File: lib/zfs.js [hunk 2]
```diff
@@ -163,8 +92,10 @@ function get(opts, cb) {
     zfs(params, function (err, stdout) {
         if (cb && typeof cb === 'function') {
             if (err) return cb(err);
-            var lines = compact(stdout.split('\n'));
-            var list = lines.map(function (x) { return new Property(x); });
+            var lines = util.compact(stdout.split('\n'));
+            var list = lines.map(function (x) { return new util.Property(x); });
+            console.log(err);
```
Comment (calmh) [inline lib/zfs.js]: These should be removed

File: lib/util.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    return Math.floor(n);
+}
+
+function Property(info) {
+    "use strict";
+
+    var obj = Object.create({});
+    if (typeof info === 'string') {
+        info = info.split(/\t+/, 4);
+        console.log(info);
```
Comment (calmh) [inline lib/util.js]: And here

**Comment by calmh** [comment]
No problem. Looks good apart from the stray debug prints; remove those, and squash to a single commit, and I think we're home. :)

**Comment by hacor** [comment]
Everything updated as you asked. Also added some guidance to every function and README is updated as well.

Sorry for the delay, it's been a very busy week.

**Comment by calmh** [comment]
No worries. It's still 18 commits though, which is a bit excessive. Squash please?

**Pull request closed without merging**